src/components/AssetDetails/AssetDetailsComponent.jsx

```diff
@@ -1,135 +1,180 @@
-import React from 'react';
+import React, { Component } from 'react';
 import PropTypes from 'prop-types';
-import { Grid, Segment, Icon, Tab, Table } from 'semantic-ui-react';
+import { isEmpty } from 'lodash';
+import { Container, Divider, Grid, Header, Segment, Icon, Tab, Table } from 'semantic-ui-react';
 import AssetAllocationHistory from '../AssetAllocationHistory';
 import AssetDescriptionComponent from '../AssetDescriptionComponent';
 import AssetNotes from '../AssetNoteComponent';
+import NavBarComponent from '../../_components/NavBarContainer';
 import LoaderComponent from '../LoaderComponent';
 
 import '../../_css/AssetDetailsComponent.css';
 
-const AssetDetailsComponent = (props) => {
-  if (props.assetLoading) {
-    return (
-      <LoaderComponent />
-    );
+class AssetDetailsComponent extends Component {
+  componentDidMount() {
+    const { assetAsigneeUsers, match, shouldFetchDetails } = this.props;
+    const { id } = match.params;
+
+    if (shouldFetchDetails) {
+      this.props.getAssetDetail(id);
+    }
+
+    if (isEmpty(assetAsigneeUsers)) {
+      this.props.loadAssetAssigneeUsers();
+    }
   }
 
-  const { assetDetail, assignedUser, errorMessage } = props;
-  const assetTabPanes = [
-    {
-      menuItem: 'Description',
-      render: () => (
-        <Tab.Pane attached={false} className="asset-tab-pane">
-          <AssetDescriptionComponent
-            {...props}
-            assetDetail={assetDetail}
-            assignedUser={assignedUser}
-            errorMessage={errorMessage}
-          />
-        </Tab.Pane>)
-    },
-    {
-      menuItem: 'Asset Note',
-      render: () => (
-        <Tab.Pane attached={false} className="asset-tab-pane current-condition">
-          <AssetNotes assetNotes={assetDetail.notes} />
-        </Tab.Pane>)
-    },
-    {
-      menuItem: 'Asset Allocation History',
-      render: () => (
-        <Tab.Pane attached={false} className="asset-tab-pane allocation-history">
-          <AssetAllocationHistory allocationHistory={assetDetail.allocation_history} />
-        </Tab.Pane>)
+  static getDerivedStateFromProps(nextProps) {
```

Comment:
This function is not needed. Please use the props directly. The state is being set and used directly without being updated using `setState`, which indicates that this function is just compromising the "signal-to-noise` ratio.

In other words, use props wherever the state variables are being utilized